collector/lib/ConfigLoader.cpp

```diff
@@ -8,114 +14,520 @@ namespace collector {
 namespace {
 const PathEnvVar CONFIG_FILE("ROX_COLLECTOR_CONFIG_PATH", "/etc/stackrox/runtime_config.yaml");
 
-enum PathTags {
+enum PathTags : uint8_t {
   LOADER_PARENT_PATH = 1,
   LOADER_CONFIG_FILE,
   LOADER_CONFIG_REALPATH,
 };
-}  // namespace
+
+std::string NodeTypeToString(YAML::NodeType::value type) {
+  // Don't add the default case so linters can warn about a missing type
+  switch (type) {
+    case YAML::NodeType::Null:
+      return "Null";
+    case YAML::NodeType::Undefined:
+      return "Undefined";
+    case YAML::NodeType::Scalar:
+      return "Scalar";
+    case YAML::NodeType::Sequence:
+      return "Sequence";
+    case YAML::NodeType::Map:
+      return "Map";
+  }
+  return "";  // Unreachable
```

Comment:
```suggestion
  return "Unknown"; 
```

I was able to reach your unreachable code using
```
CLOG(INFO) << "NodeTypeToString= " << NodeTypeToString(static_cast<YAML::NodeType::value>(17));
```

The output was

[INFO    2025/02/13 02:22:43] NodeTypeToString=

Reply:
This is odd, I'll have to try it, I'm fairly certain I covered all the cases for `YAML::NodeType`.

Reply:
I think I am just being a bit nit picky. I think in practice it is unlikely that part of the code would be reached, but it is possible that either a new YAML::NodeType enum is introduced or someone writes some strange code like I did. However, I prefer that the final return be something other than an empty string and I would not comment code as being unreachable unless it was inside something like `if (false)`. But then why would I have that?

Reply:
I didn't read this comment properly the first time around. This is not the way to use an enum, the whole reason enums exist is to give a limited, well defined set of values that can be used at any given time, if you are going to break that invariant then there's no code I can write that will satisfy every possible condition anyone could come up with. 

If you use the enum the way it is intended to be used, then that last line is effectively unreachable because all possible values have been accounted for in the switch-case: https://github.com/jbeder/yaml-cpp/blob/f7320141120f720aecc4c32be25586e7da9eb978/include/yaml-cpp/node/type.h#L11-L13

Now, why is the last `return` value there? Because the compiler will complain that the function reaches the end without a return statement because it is not smart enough to figure out that it is unreachable. Could I use a `default` branch in the switch case? Yes, but the advantage of doing it this way is that if you have proper linting setup for the project, (which I've wanted to implement for some time now) and you miss a value, you'll see something like this:
![image](https://github.com/user-attachments/assets/86fc0265-7894-4c7a-9cce-2f2b06cf801e)

This in turn means that if we update `yaml-cpp` in the future and an enum value is added, we would know and could act on it. If the `default` case is added though, the lint check goes away, defeating the purpose.